### PR TITLE
Denobyexample

### PR DESCRIPTION
**Pull request by thisisjofrank**

### CODE REVIEW

**Comment by lucacasonato** [comment]
dark mode needs another pass: 
image

**Comment by lucacasonato** [comment]
scroll bars on mobile are not great:

image


Why the change with the space between code blocks on desktop? They are meant to be continuous, because the notes are just annotations on a continuous block of code (for example https://examples.deno.land/http-server-routing).

File: src/components/DenoByExample/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                        <h1 className="text-4xl md:text-6xl">Deno by Example</h1>
+                        <p className="max-w-prose mx-auto">
+                            A collection of annotated Deno examples, to be used as a reference for how to build with Deno, or as a guide to learn about many of Deno's features.
+                        </p>
+                    </div>
+                    <div className="columns-1 sm:columns-2 lg:columns-3 xl:columns-4" style={{ columnGap: '3rem'}}>
+                        {elements}
+                    </div>
+                </div>
+            </div>
```
Comment (ry) [inline src/components/DenoByExample/index.tsx:86]: Add a CTA for adding examples

**Comment by thisisjofrank** [comment]
> scroll bars on mobile are not great:
> 
> image
> 
> Why the change with the space between code blocks on desktop? They are meant to be continuous, because the notes are just annotations on a continuous block of code (for example https://examples.deno.land/http-server-routing).

Argh, somehow didn't spot that this change had crept in. I've updated but had to do some CSS crimes to fight docusaurus' code block styling.

**Comment by lucacasonato** [comment]
Functionality looks good now! Thanks :)

**Review by ry — APPROVED** [review]
LGTM

File: src/components/DenoByExample/ByExample.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    tags,
+    additionalResources,
+    run,
+    playground,
+    files,
+  } = parsed;
+
+  const freshProps = {
+    url: {
+      origin: "https://github.com/denoland/deno-docs/blob/by-example/",
```
Comment (kt3k) [inline src/components/DenoByExample/ByExample.tsx:26]: It seems the branch name is needed before the path?

```suggestion
      origin: "https://github.com/denoland/deno-docs/blob/main/by-example/",
```

**Review by kt3k — COMMENTED** [review]
Left one comment. Other part looks great 👍

**Pull request merged**